gradle/deploy.gradle

```diff
@@ -62,8 +62,29 @@ configure(subprojects - [project(':util'), project(':driver-legacy')]) {
 
     task uploadSnapshotArchives(dependsOn: uploadArchives) {
         if (!version.endsWith("-SNAPSHOT")) {
-            enabled = false;
-            dependsOn = [];
+            enabled = false
+            dependsOn = []
+        }
+    }
+
+    task publishArchives(dependsOn: uploadArchives) {
+        String gitVersion = getGitVersion()
+        if (gitVersion != version) {
+           dependsOn = []
+        }
+        doFirst {
+            if (gitVersion != version) {
+                def cause = """
+                | Version mismatch:
+                | =================
+                |
+                | $version != $gitVersion 
+                |
+                | The project version does not match the ClientMetaData version.
```

Comment:
The comparison is between the output of `git describe` and the value of the `version` property in build.gradle.  ClientMetaData is kind of orthogonal to this check.  How about "The project version does not match the git tag"?

Reply:
Yeah makes more sense.